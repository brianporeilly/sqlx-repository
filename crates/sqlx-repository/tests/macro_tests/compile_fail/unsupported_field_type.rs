//! Test that repository derive fails with unsupported field types

use sqlx_repository::prelude::*;
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow, Repository)]
#[repository(table = "users")]
pub struct User {
    pub id: i32,
    pub name: String,
    pub metadata: HashMap<String, String>,  // Unsupported type
}

fn main() {}