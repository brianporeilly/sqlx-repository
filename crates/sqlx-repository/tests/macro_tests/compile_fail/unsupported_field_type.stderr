error: Unsupported field type 'HashMap' for field 'metadata'.

       Supported types:
       - Integers: i16, i32, i64, u16, u32, u64
       - Floats: f32, f64
       - Text: String, &str
       - Boolean: bool
       - Time: DateTime<Utc>, Date, Time
       - Optional: Option<T> for any supported type T
       - Collections: Vec<T> for supported types T
       - UUID: Uuid (with uuid feature)

       For complex types, consider using JSON serialization with String fields.
  --> tests/macro_tests/compile_fail/unsupported_field_type.rs:12:19
   |
12 |     pub metadata: HashMap<String, String>,  // Unsupported type
   |                   ^^^^^^^^^^^^^^^^^^^^^^^
